### PR TITLE
Fix 4622: Add Bridge theme static file exclusions

### LINKED ISSUE #4622: Automatically exclude Bridge theme files from static file generation
**Before submitting an issue please check that you’ve completed the following steps:**
- Made sure you’re on the latest version - **Yes, 3.10.5.1**
- Used the search feature to ensure that the bug hasn’t been reported before - **Yes**

**Describe the bug**
Static File Creation breaks [Bridge theme's](http://bridge.qodeinteractive.com/) styles and scripts.

**To Reproduce**
1. Install Bridge theme (not sure if any specific Bridge feature triggered the error).
2. Enable WP Rocket with no UI features enabled. 
3. See broken layout and errors in the console:
image

**Expected behavior**
Automatically exclude these Bridge theme files from the static file generation optimization.

**Additional context**
The issue is fixed if these files are excluded using the [WP Rocket | Exclude from Static File Creation](https://github.com/wp-media/wp-rocket-helpers/raw/master/static-files/wp-rocket-static-exclude-dynamic-files/wp-rocket-static-exclude-dynamic-files.zip) helper plugin.

```
wp-content/themes/bridge/js/default_dynamic_callback.php
wp-content/themes/bridge/css/style_dynamic_callback.php
```

Slack thread: https://wp-media.slack.com/archives/C08N8J6VC/p1640797019014100
Tickets: https://secure.helpscout.net/conversation/1737671894/315972/ and https://secure.helpscout.net/conversation/1742613768/316567/

**Backlog Grooming (for WP Media dev team use only)**
- [ ] Reproduce the problem
- [ ] Identify the root cause
- [ ] Scope a solution
- [ ] Estimate the effort

### PR DESCRIPTION
**Pull request by iCaspar**
## Description

This adds a new method to the `ThirdParty/Bridge` subscriber: `exclude_static_file_generation(array $files): array`.
The new method is a filter hooked to `rocket_exclude_static_dynamic_resources` that will add certain static files generated by Bridge in some edge cases. (See discussion on #4622.)

Corresponding integration test is included.

Fixes #4622

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## Is the solution different from the one proposed during the grooming?

Same solution outlined in the grooming notes.

## How Has This Been Tested?
Integration test added to confirm filter includes specified Bridge-generated files.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: inc/ThirdParty/Themes/Bridge.php [hunk 1]
```diff
@@ -94,4 +93,18 @@ public function maybe_clear_cache( $old_value, $new_value ) {
...
+	/**
+	 * Excludes Bridge statically generated files.
+	 *
+	 * @param array $files Array of static files to be excluded.
+	 *
+	 * @return array Excluded files with Bridge entries to be excluded.
+	 */
+	public function exclude_static_file_generation( $files ) {
+		$files[] = 'wp-content/themes/bridge/js/default_dynamic_callback.php';
+		$files[] = 'wp-content/themes/bridge/css/style_dynamic_callback.php';
```
Comment (remyperona) [inline inc/ThirdParty/Themes/Bridge.php]: I'd suggest to get the path more dynamically here, since the `wp-content` folder can be a different name.

There is an example here: https://github.com/wp-media/wp-rocket/blob/e2c615669d775508771c0bcfa40eef9d69068265/inc/ThirdParty/Themes/Avada.php#L133

**Review by remyperona — CHANGES_REQUESTED** [review]

**Review by remyperona — APPROVED** [review]

**Comment by piotrbak** [comment]
@iCaspar This function is not called, seems that it's not working on our testing environment:
https://github.com/wp-media/wp-rocket/blob/e244104666ad06ca020753d3b36f1dbbea76319d/inc/ThirdParty/Themes/Bridge.php#L106

Don't know what's the reason, but the only exclusion here is ajax call:
https://github.com/wp-media/wp-rocket/blob/e244104666ad06ca020753d3b36f1dbbea76319d/inc/Engine/Optimization/CacheDynamicResource.php#L84

**Comment by remyperona** [comment]
@piotrbak on the testing environment, is the theme name "Bridge"?
